cpp/include/cudf/transform.hpp

```diff
@@ -140,6 +140,31 @@ std::pair<std::unique_ptr<cudf::table>, std::unique_ptr<cudf::column>> encode(
   cudf::table_view const& input,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Encode the labels in column with an indicator function
+ *
+ * The `i`th row of the `j`th column in the output table equals 1
+ * if input_column[i] == categories[j], and 0 otherwise.
+ *
+ * Examples:
+ * @code{.pseudo}
+ * input: [{'a', 'c', null, 'c', 'b'}]
+ * categories: ['c', null]
```

Comment:
Is there ever a use case where we don't want the categories to be the full set of unique elements in input?

Reply:
Isn't this example showing categories != full set of unique(input)? Maybe I'm missing sth?

Reply:
Right, I'm asking whether this example is a realistic usage of the function. Basically I'm wondering why the categories parameter actually needs to exist.

Reply:
Good question.. From ML libraries the answer to this doesn't seem ubiquitous. The tensorflow one_hot op assumes all categories are encoded. The pytorch `nn.function.one_hot` allows an extra argument `num_classes` to control the number of categories in the output. And scikit-learn's OneHotEncoder allows user manually specifying the number of categories in output. 

Our `Series.one_hot_encoding` allows user to specify the output categories - and this is basically where I got the intuition from. But this is not a standard pandas API and we should be able to change that. Not sure if this breaks any existing workflow though.

Reply:
This is maybe a good question for @beckernick, or he might know who else would care. I'm just not sure since I have personally never had a use for one-hot encoding that excludes some categories.

Reply:
In the distributed setting, one partition may have keys {"a","b","c"} and another has {"b","c","d"}.

In such a scenario, my understanding is that we cannot rely on a local-only mapping of key to category code, as we will end up with collisions (i.e., does "a" or "b" map to 0). Instead, we first compute a global set of categories, and broadcast the global mapping to each local partition.

I believe Dask does this by: 1) creating a dictionary of the global keys on the Client process; 2) broadcasting the global dictionary to each worker; 3) doing a within-partition categorical conversion using the global categories.

Reply:
Ah, I may have misunderstood the context on quick read. I'm also unfamiliar with a scenario in which someone would want to one-hot encode only a subset of the categories.

Reply:
Perhaps in the event of an expected large cardinality, we don't want to generate columns for each of those.

Reply:
Yeah that is possible. It's easier to add a parameter than to remove one, so if we aren't sure that such a need exists my preference would be to start with the minimal API and add this parameter later (we could always use some sort of sentinel to indicate "use all unique elements"), but if we expect this to be useful at some point perhaps we're better off leaving this API as is. I just want to make sure we've considered this carefully before finalizing the API.

Reply:
If not specifying the category column, what should be order of the categories of the output columns? Should it just be the `unique` of this column? 

Or the sorted column of the uniques? Should nulls and nans be before or after the sorted version?

Reply:
Having discussed this on Slack, we've decided to stick with the API specifying categories.